File: app/src/main/java/pl/adriankremski/coolector/authentication/login/LoginMvp.java
package pl.adriankremski.coolector.authentication.login;

import io.reactivex.disposables.Disposable;

public interface LoginMvp {

    interface View {
        void registerDisposable(Disposable disposable);
        void showLoading();
        void hideLoading();
        void showNetworkError();
        void showLoginSuccess();
        void showMainScreen();
        void closeScreen();
    }

    interface Presenter{
        void onCreate();
        void loginWithEmail(String email, String password);
    }
}

File: app/src/main/java/pl/adriankremski/coolector/authentication/login/LoginMvp.kt
package pl.adriankremski.coolector.authentication.login

import io.reactivex.disposables.Disposable

interface LoginMvp {

    interface View {
        fun registerDisposable(disposable: Disposable)
        fun showLoading()
        fun hideLoading()
        fun showNetworkError()
        fun showLoginSuccess()
        fun showMainScreen()
        fun closeScreen()
    }

    interface Presenter{
        fun onCreate()
        fun loginWithEmail(email: String, password: String)
    }
}
